### DISCUSSION TITLE
Please add a huge icon for "Filter is active"

### DISCUSSION THREAD

**Discussion by Pierre-de-la-Verre** [WinMerge/winmerge#1056] [Ideas]
A typical user problem
- I compared two special files and activated the "line filter"
- Went home, forgot the filter
- Compared two standard files - got nearly no differences
- Why: I forgot that the filter was active.

So: Please add an eye-catching information (icon, red text, modified file-tab) to make it visible.

**Comment by sdottaka** [comment]
The commit 3c84e8b5936562e7b87e9376dcaa3f32afbeb3be now adds an (F) to the title bar and tab bar if there are lines that are ignored by the line filters, substitution filters, and prediffer plugins.
![image](https://user-images.githubusercontent.com/98126/143460307-a8a719a0-8417-4e9c-b4bc-333f2efacc21.png)

**Comment by Pierre-de-la-Verre** [comment]
Great!

**Comment by Gitoffthelawn** [comment]
I just noticed this improvement.  Is there a tooltip for the tab with the `(F)` that explains what it means?  If not, probably a good idea.

BTW, I'll be testing the new beta that includes the functionality added by this issue report within 24 hours (hopefully!).

**Comment by sdottaka** [comment]
> Is there a tooltip for the tab with the (F) that explains what it means? 

Unfortunately, the tooltip does not currently show its meaning.

**Reply by Gitoffthelawn** [reply]
Would it be reasonable to add?  I think people who only occasionally use filters will not know/remember what that `(F)` means.

**Comment by sdottaka** [comment]
I think it's a good idea.

However, so far I haven't come up with a smart way to implement it. (I think it is possible to implement it in a muddled way).

**Comment by goozak** [comment]
That (F) is not shown when comparing folders.

Maybe having an option to NOT save the Filter between executions would be helpful.  When I want to use a filter, I usually create a project for that comparison.  I don't need to have 'sticky' filters.

**Reply by Gitoffthelawn** [reply]
> Maybe having an option to NOT save the Filter between executions would be helpful. When I want to use a filter, I usually create a project for that comparison. I don't need to have 'sticky' filters.

This sounds like a good idea.  A possible implementation could be to allow the user to enable/disable filters for this session only, and to enable/disable filters as the default when a WinMerge is initiated.

Perhaps it could be something as simple as a "Change settings for this session only" checkbox/button for the `Filters` dialog box.

**Comment by Pierre-de-la-Verre** [comment]
> That (F) is not shown when comparing folders.

It is not shown either when ZIPs are compared and the filter is active in the background.
